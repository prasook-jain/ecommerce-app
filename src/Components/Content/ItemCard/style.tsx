import styled from "styled-components";

export const ItemDiv = styled.div`
  display: flex;
  flex-direction: column;

  &:hover {
    box-shadow: 0 0 3px 3px lightgray;
  }

  & .item-image {
    width: 330px;
  }

  & .item-image-container {
    display: flex;
    justify-content: center;
    height: 400px;
    align-items: center;
    overflow: hidden;
    background: white;
  }

  & .item-details {
    box-sizing: border-box;
    padding: 1rem 1rem;
    display: grid;
    grid-template-areas:
      "item-name item-name"
      "item-price cart-button";

    height: 100px;
    border-top: 1px solid black;
    background-color: white;

    & > .item-name {
      font-size: 14px;
      line-height: 18px;
      height: 36px;
      font-weight: 700;
      grid-area: item-name;
    }

    & > .item-price {
      display: flex;
      align-items: center;
      grid-area: item-price;
      div {
        margin-right: 1rem;
      }
      .item-actual-price {
        color: darkgray;
        text-decoration: line-through;
      }
    }

    & > .cart-button {
      color: white;
      padding: 0.2rem;
      border-radius: 4px;
      background-color: blue;
      grid-area: cart-button;

      :hover {
        cursor: pointer;
      }
    }
  }
`;
