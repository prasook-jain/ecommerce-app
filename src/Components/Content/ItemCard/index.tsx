import { useDispatch } from "react-redux";
import { ItemDiv } from "./style";
import { ADD_ITEM_TO_CART } from "../../../reduxStore/action";
import { IItem } from "../../../utility/types";

export interface IItemCardProps {
  item: IItem;
}

const ItemCard: React.FunctionComponent<IItemCardProps> = (props) => {
  const { id, name, price, image_urls, discount_price } = props.item;

  const dispatch = useDispatch();

  const handleAddCartButton = () => {
    dispatch({
      type: ADD_ITEM_TO_CART,
      payload: {
        cartItem: { itemId: id, quantity: 1 },
      },
    });
  };

  const itemName = name.length > 75 ? name.slice(0, 75) + "..." : name;
  return (
    <ItemDiv>
      <div className="item-image-container">
        <img className="item-image" src={image_urls[0]} alt={name} />
      </div>
      <div className="item-details">
        <div className="item-name" title={name}>
          {itemName}
        </div>
        <div className="item-price">
          <div className="item-discount-price">₹{discount_price}</div>
          <div className="item-actual-price">₹{price}</div>
        </div>
        <button className="cart-button" onClick={handleAddCartButton}>
          Add to Cart
        </button>
      </div>
    </ItemDiv>
  );
};

export default ItemCard;
