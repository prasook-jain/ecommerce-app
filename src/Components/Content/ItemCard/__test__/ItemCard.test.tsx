import React from "react";
import mockBackend from "../../../../utility/mock.json";

import { render, screen } from "@testing-library/react";
import ItemCard, { IItemCardProps } from "..";

const mockProps = {
  item: mockBackend["itemsHash"].BDDEPW5FY7FGRGEZ,
};

const mockDispatch = jest.fn();
jest.mock("react-redux", () => {
  useDispatch: () => mockDispatch;
});

describe("ItemCard : ", () => {
  test("renders correctly, snapshot test", () => {
    const { container } = render(<ItemCard item={mockProps.item} />);
    expect(container).toMatchSnapshot();
  });
});
