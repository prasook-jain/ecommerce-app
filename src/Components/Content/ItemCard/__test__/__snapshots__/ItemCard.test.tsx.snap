// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard :  renders correctly, snapshot test 1`] = `
<div>
  <div
    class="sc-bdfBwQ inRgkj"
  >
    <div
      class="item-image-container"
    >
      <img
        alt="Spacewood Mayflower Engineered Wood Queen Box Bed"
        class="item-image"
        src="https://rukminim1.flixcart.com/image/312/312/jk8lz0w0/bed/g/e/z/queen-na-particle-board-kosmo-mayflower-queen-bed-box-spacewood-original-imaf7mtfdv8hfg6z.jpeg?q=70"
      />
    </div>
    <div
      class="item-details"
    >
      <div
        class="item-name"
        title="Spacewood Mayflower Engineered Wood Queen Box Bed"
      >
        Spacewood Mayflower Engineered Wood Queen Box Bed
      </div>
      <div
        class="item-price"
      >
        <div
          class="item-discount-price"
        >
          ₹
          17790
        </div>
        <div
          class="item-actual-price"
        >
          ₹
          31140
        </div>
      </div>
      <button
        class="cart-button"
      >
        Add to Cart
      </button>
    </div>
  </div>
</div>
`;
